rust/azure_iot_operations_services/tests/azure_device_registry_client_tests.rs

```diff
@@ -0,0 +1,623 @@
+        )
+        .is_ok()
+    );
+}
+
+#[tokio::test]
+// #[ignore = "This test is ignored as it is still failing."]
+async fn observe_device_update_notifications() {
+    let log_identifier = "observe_device_update_notifications_network_tests-rust";
+    if !setup_test(log_identifier) {
+        return;
+    }
+    let (session, azure_device_registry_client, exit_handle) =
+        initialize_client(&format!("{log_identifier}-client"));
+
+    let test_task = tokio::task::spawn({
+        async move {
+            let mut observation = azure_device_registry_client
+                .observe_device_update_notifications(
+                    DEVICE1.to_string(),
+                    ENDPOINT1.to_string(),
+                    TIMEOUT,
+                )
+                .await
+                .unwrap();
+            log::info!("[{log_identifier}] Device update observation: {observation:?}",);
+            let receive_notifications_task = tokio::task::spawn({
+                async move {
+                    log::info!("[{log_identifier}] Device update notification receiver started.");
+                    let mut count = 0;
+                    if let Some((device, _)) = observation.recv_notification().await {
+                        count += 1;
+                        log::info!("[{log_identifier}] FIRST OBS DELETE LOG");
+                        log::info!("[{log_identifier}] Device Observation expected: {device:?}");
+                        assert_eq!(device.name, DEVICE1);
+                    }
+                    while let Some((device, _)) = observation.recv_notification().await {
+                        count += 1;
+                        log::info!("[{log_identifier}] ANY OTHER OBS DELETE LOG");
+                        log::info!("[{log_identifier}] Device Observation unexpected: {device:?}");
+                        // if something weird happens, this should prevent an infinite loop.
+                        // Note that this does prevent getting an accurate count of how many extra unexpected notifications were received
+                        assert!(count < 2);
+                    }
+                    // only the 1 expected notifications should occur
+                    assert_eq!(count, 1);
+                    log::info!("[{log_identifier}] Device update notification receiver closed");
+                }
+            });
+
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            let response = azure_device_registry_client
+                .get_device(DEVICE1.to_string(), ENDPOINT1.to_string(), TIMEOUT)
+                .await
+                .unwrap();
+            log::info!("[{log_identifier}] Get device to update the status: {response:?}",);
+            let mut endpoint_statuses = HashMap::new();
+            for (endpoint_name, endpoint) in response.specification.endpoints.inbound {
+                if endpoint.endpoint_type == ENDPOINT_TYPE {
```

Comment:
will this always be true? I wouldn't add conditional handling logic within a test, the test should always behave the same way